frontend-react/src/layouts/Markdown/MarkdownLayout.tsx

```diff
@@ -0,0 +1,267 @@
+import MDXModules from "../../MDXModules";
+
+import { TableOfContents } from "./TableOfContents";
+import { CallToAction } from "./CallToAction";
+import MarkdownLayoutContext from "./Context";
+import { LayoutSidenav, LayoutMain } from "./LayoutComponents";
+import styles from "./MarkdownLayout.module.scss";
+
+/**
+ * React components are functions that are pascal-cased so filtering is done
+ * so.
+ */
+function filterComponents<T extends {}>(
+    obj: T,
+    include: Array<string & keyof T> = []
+) {
+    return Object.fromEntries(
+        Object.entries(obj).filter(
+            ([k, v]) =>
+                include.includes(k as string & keyof T) ||
+                (typeof v === "function" && k[0] === k[0].toUpperCase())
+        )
+    );
+}
+
+const uswdsComponents = filterComponents(reactUSWDS);
+const sharedComponents = filterComponents(shared);
+
+export interface MarkdownLayoutProps {
+    children: JSX.Element;
+    frontmatter?: {
+        sidenav?: string;
+        breadcrumbs?: Array<{ label: string; href: string }>;
+        title?: string;
+        subtitle?: string | string[];
+        callToAction?: Array<{ label: string; href: string }>;
+        lastUpdated?: string;
+        toc?: boolean | { depth?: number };
+        backToTop?: boolean;
+    };
+    toc?: TocEntry[];
+    main?: JSX.Element;
+    nav?: JSX.Element;
+    mdx?: React.ComponentProps<typeof MDXProvider>;
+}
+
+/**
+ * Markdown-formatted elements use lowercase variants from this list. If you need manual
+ * control over props, use pascal-cased variants. react-uswds components dynamically added
+ * to list (overridden by project shared components).
+ */
+const MDXComponents = {
+    a: USSmartLink,
+    A: USSmartLink,
+    ...uswdsComponents,
+    ...sharedComponents,
+    Link: USSmartLink,
+    USNavLink,
+    Subtitle: (props: React.PropsWithChildren<{}>) => {
+        return <p>{props.children}</p>;
+    },
+};
+
+/**
+ * Provides a LayoutSidenav component in mdx context that when used will automatically
+ * enable the sidenav of layout.
+ *
+ * FUTURE_TODO: Remove GridContainer once implemented in higher-level component.
+ *
+ * Default markdown layout.
+ * @example
+ * With sidenav:
+ * +------+-----------------+
+ * |      |                 |
+ * |      |                 |
+ * | SIDE |      MAIN       |
+ * |      |                 |
+ * |      |                 |
+ * +------+-----------------+
+ * Without sidenav:
+ * +------------------------+
+ * |                        |
+ * |                        |
+ * |         MAIN           |
+ * |                        |
+ * |                        |
+ * +------------------------+
+ */
+export function MarkdownLayout({
+    children,
+    main,
+    mdx,
+    frontmatter: {
+        title,
+        breadcrumbs,
+        subtitle,
+        callToAction,
+        lastUpdated,
+        toc,
+        backToTop,
+    } = {},
+    toc: tocEntries,
+}: MarkdownLayoutProps) {
+    const helmet = title ? (
```

Comment:
This feels small enough that we don't need to declare a variable here and just put it directly: https://github.com/CDCgov/prime-reportstream/pull/10728/files#diff-b3c320aa23eac426c39fd13d10d2bdfa456981571b886597797b241d19dc51d2R157 

Also, since there's uppercase `Helmet` being used here, having a lowercase `helmet` also declared feels a little confusing if we don't have to.

Reply:
Agreed, fixed.